### PR TITLE
Adds readme for more extension items

### PR DESCRIPTION
**Pull request by maryamariyan**
Related to: #59630, #77413, #77733

### CODE REVIEW

File: src/libraries/Microsoft.Extensions.Caching.Memory/README.md [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+
+Documentation can be found at https://learn.microsoft.com/en-us/dotnet/core/extensions/caching.
+
+## Contribution Bar
+- [x] [We consider new features, new APIs, bug fixes, and performance changes](../../libraries/README.md#primary-bar)
+
+The APIs and functionality need more investment in the upcoming .NET releases.
+
+## Deployment
+[Microsoft.Extensions.Caching.Memory](https://www.nuget.org/packages/Microsoft.Extensions.Caching.Memory) is not included in the shared framework. The package is deployed as out-of-band (OOB) and needs to be installed into projects directly.
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Caching.Memory/README.md:13]: >is not included in the shared framework [](http://example.com/codeflow?start=106&length=39)

is included in the ASP.NET Core shared framework.

File: src/libraries/Microsoft.Extensions.Configuration.Abstractions/README.md [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+
+Documentation can be found at https://learn.microsoft.com/en-us/dotnet/core/extensions/configuration.
+
+## Contribution Bar
+- [x] [We consider new features, new APIs, bug fixes, and performance changes](../../libraries/README.md#primary-bar)
+
+The APIs and functionality are mature, but do get extended occasionally.
+
+## Deployment
+[Microsoft.Extensions.Configuration.Abstractions](https://www.nuget.org/packages/Microsoft.Extensions.Configuration.Abstractions) is not included in the shared framework. The package is deployed as out-of-band (OOB) and needs to be installed into projects directly.
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Configuration.Abstractions/README.md:13]: >is not included in the shared framework [](http://example.com/codeflow?start=130&length=39)

is included in the ASP.NET Core shared framework.

Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Configuration.Abstractions/README.md:13]: >is not included in the shared framework. The package is deployed as out-of-band (OOB) and needs to be installed into projects directly. [](http://example.com/codeflow?start=130&length=135)

is included in the ASP.NET Core shared framework. The package is deployed as out-of-band (OOB) too and can be referenced into projects directly.

File: src/libraries/Microsoft.Extensions.Caching.Memory/README.md (hunk 1 above)
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Caching.Memory/README.md:13]: >is not included in the shared framework. The package is deployed as out-of-band (OOB) and needs to be installed into projects directly. [](http://example.com/codeflow?start=106&length=135)

is included in the ASP.NET Core shared framework. The package is deployed as out-of-band (OOB) too and can be referenced into projects directly.

File: src/libraries/Microsoft.Extensions.Configuration.Abstractions/README.md [hunk 3]
```diff
@@ -0,0 +1,13 @@
+# Microsoft.Extensions.Configuration.Abstractions
```
Comment (tarekgh) [inline src/libraries/Microsoft.Extensions.Configuration.Abstractions/README.md:1]: This file already added by @MSDN-WhiteKnight I guess

Comment (MSDN-WhiteKnight) [reply src/libraries/Microsoft.Extensions.Configuration.Abstractions/README.md:1]: Yes, in this commit: https://github.com/dotnet/runtime/pull/77733/files#diff-46f803b9b3f17e6cbf7cc2d229bd3e906e08ecf82eb7b2a21fe736cbb2637818 That's why this PR shows conflicts.

**Review by tarekgh — APPROVED** [review]

**Pull request merged**